### PR TITLE
Fix/remove skipped test cases

### LINKED ISSUE #470: Remove Skipped Tests and Ensure Test Suite is Running Fully
### Issue Description

Remove all instances of skipped tests in the codebase (tests marked with .skip or similar), and ensure that all tests are running as part of the test suite. This task also includes identifying any issues or failures with previously skipped tests and fixing them if necessary.

**Locate Skipped Tests:**

Search for any tests in the codebase that are marked as skipped (e.g., it.skip, test.skip, describe.skip, etc.).
Remove Skip Flags:

Remove .skip or equivalent flags from all test cases.

**Run the Test Suite:**

Execute the entire test suite using the appropriate test command (e.g., npm test, pnpm test, yarn test).

**Address Failing Tests:**

Investigate and fix any test failures that occur after re-enabling the skipped tests.
Verify Successful Test Run:

Ensure that all tests pass and the suite completes without any skips.
Update Documentation (if needed):

Ensure that test-related documentation reflects any updates or changes, including the resolution of previously skipped tests.### Screenshots

[If applicable, add screenshots or images that help illustrate the issue. You can use the following Markdown syntax to embed images:]

### Environment

- Operating System: [e.g., Windows, macOS, Linux]
- Browser (if applicable): [e.g., Chrome, Firefox, Safari]
- Version of Software/Project: [e.g., v1.0.0]

### Additional Information

[Include any additional context, logs, or details that might be helpful in addressing the issue.]

### Checklist

- [ ] I have read and followed the project's code of conduct.
- [ ] I have searched for similar issues before creating this one.
- [ ] I have provided all the necessary information to understand and reproduce the issue.
- [ ] I am willing to contribute to the resolution of this issue.

---

Thank you for contributing to our project! We appreciate your help in improving it.

📚 See [contributing instructions](https://github.com/samyakshah3008/mobile-app/blob/develop/CONTRIBUTING.md).

### PR DESCRIPTION
**Pull request by abhisheksharmayt**
Date: 02/11/2024

Developer Name: Abhishek Sharma

---
- Closes #470 

## Description

1. Removed skipped test cases for the currently used components.
2. I updated some test cases for components whose features were updated, but the test cases were skipped.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>
</summary>

</details>

## Test Coverage
<details>
<summary>Before</summary>

image

</details>
<details>
<summary>After</summary>

image

</details>

### CODE REVIEW

**Review by ashifkhn — DISMISSED** [review]

File: __tests__/Goals/components/Create-Goals-test.tsx [hunk 1]
```diff
@@ -59,13 +65,57 @@ describe('MainScreen', () => {
...
+
+    await waitFor(() => {
+      expect(Alert.alert).toHaveBeenCalledWith(
+        'Success',
+        `Task has been created and assigned to test user`,
+        [{ text: 'OK', onPress: expect.any(Function) }],
+      );
+    });
+
+    // @ts-ignore
```
Comment (vinit717) [inline __tests__/Goals/components/Create-Goals-test.tsx]: Why did you ignore the typescript check?

Comment (abhisheksharmayt) [reply __tests__/Goals/components/Create-Goals-test.tsx]: To access the onPress button on the system alert. I referred to this https://github.com/callstack/react-native-testing-library/issues/218

Comment (vinit717) [reply __tests__/Goals/components/Create-Goals-test.tsx]: Can you tell the error you are facing when removing this ts ignore command

Comment (abhisheksharmayt) [reply __tests__/Goals/components/Create-Goals-test.tsx]: image

Comment (abhisheksharmayt) [reply __tests__/Goals/components/Create-Goals-test.tsx]: I can get rid of the ts error by asserting 
spyAlert.mock.calls?.[0]?.[2]?.[0].onPress!();

Comment (vinit717) [reply __tests__/Goals/components/Create-Goals-test.tsx]: Try to make it like this
[spyAlert.mock.calls?.[0]?.[2]?.[0]?.onPress](https://github.com/callstack/react-native-testing-library/issues/218#issuecomment-2117747532)

Comment (abhisheksharmayt) [reply __tests__/Goals/components/Create-Goals-test.tsx]: image

need to execute onPress.

```spyAlert.mock.calls?.[0]?.[2]?.[0].onPress!();```
The above is working, should I go ahead with this?

Comment (vinit717) [reply __tests__/Goals/components/Create-Goals-test.tsx]: call it here like this onPress()

Comment (abhisheksharmayt) [reply __tests__/Goals/components/Create-Goals-test.tsx]: image

Comment (vinit717) [reply __tests__/Goals/components/Create-Goals-test.tsx]: do this then 
spyAlert.mock.calls?.[0]?.[2]?.[0].onPress!();

or the one mention in that link

Comment (abhisheksharmayt) [reply __tests__/Goals/components/Create-Goals-test.tsx]: ```spyAlert.mock.calls?.[0]?.[2]?.[0].onPress!();```

This is working as I have mentioned already in previous, should I go ahead with this?

Comment (vinit717) [reply __tests__/Goals/components/Create-Goals-test.tsx]: Yes and by using this does the tests are passing?

Comment (abhisheksharmayt) [reply __tests__/Goals/components/Create-Goals-test.tsx]: Yes

**Review by vinit717 — DISMISSED** [review]

File: __tests__/Goals/components/Create-Goals-test.tsx [hunk 2]
```diff
@@ -59,13 +68,56 @@ describe('MainScreen', () => {
...
+
+    await waitFor(() => {
+      expect(Alert.alert).toHaveBeenCalledWith(
+        'Success',
+        `Task has been created and assigned to test user`,
+        [{ text: 'OK', onPress: expect.any(Function) }],
+      );
+    });
+
+    spyAlert.mock.calls?.[0]?.[2]?.[0].onPress!();
```
Comment (MehulKChaudhari) [inline __tests__/Goals/components/Create-Goals-test.tsx]: Nit: could we please make a variable with good name and then call the `onPress!()` 

For eg: const spyAlertMockCall = spyAlert.mock.calls?.[0]?.[2]?.[0];

This is for better readability of code.

Comment (abhisheksharmayt) [reply __tests__/Goals/components/Create-Goals-test.tsx]: Sure, I'll update.

**Review by MehulKChaudhari — DISMISSED** [review]
Looks to good me! 

Thanks for working on it

File: __tests__/AuthScreen-test.tsx [hunk 3]
```diff
@@ -13,22 +14,24 @@ jest.mock('react-redux', () => {
...
-      <Toast />
-    </>,
-  );
+it('Clicking on Sign in with Github opens browser', async () => {
+  customRenderWithProvider(AuthScreen);
+
   const githubSignInBtn = screen.getByText(Strings.SIGN_IN_BUTTON_TEXT);
   fireEvent.press(githubSignInBtn);
-  screen.getByText(/Sign in with GitHub coming soon/i);
+  expect(Linking.openURL).toHaveBeenCalledTimes(1);
```
Comment (joyguptaa) [inline __tests__/AuthScreen-test.tsx:57]: Can you change `toHaveBeenCalledTimes` with `toHaveBeenCalledWith` and ensure that this has been called with the correct link

Comment (abhisheksharmayt) [reply __tests__/AuthScreen-test.tsx:57]: updated @joyguptaa

**Review by ashifkhn — DISMISSED** [review]

File: src/utils/tests/ProviderWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import { store } from '../../../App';
+
+interface ProviderWrapperProps {
+  children: ReactNode;
+}
+
+const ProviderWrapper: React.FC<ProviderWrapperProps> = ({ children }) => (
+  <StoreProvider store={store}>
+    <AuthProvider>
+      <>{children}</>
```
Comment (vinit717) [inline src/utils/tests/ProviderWrapper.tsx]: Why this need to wrap in fragment?

Comment (abhisheksharmayt) [reply src/utils/tests/ProviderWrapper.tsx]: image

AuthProvider children type is defined as JSX.Element to allow ( null | undefined) did this.

Comment (vinit717) [reply src/utils/tests/ProviderWrapper.tsx]: why we are allows AuthProvider to be  null or undefined as children

Comment (abhisheksharmayt) [reply src/utils/tests/ProviderWrapper.tsx]: I don't remember exactly why I did this, removed.
https://github.com/Real-Dev-Squad/mobile-app/pull/477/files#diff-57cdfd62a0137f975515ef87ba1e94e78cc9fa0bfc6d1f215e3105896b4c1fccR11-R14

**Review by vinit717 — APPROVED** [review]

**Review by ashifkhn — APPROVED** [review]

**Pull request merged**